app/services/concerns/measure_unit_presentable.rb

```diff
@@ -13,9 +14,15 @@ def applicable_unit
     applicable_units[component.unit]
   end
 
+  def unit_answer_for(unit)
+    key = unit['coerced_measurement_unit_code'].presence || "#{unit['measurement_unit_code']}#{unit['measurement_unit_qualifier_code']}"
+
+    user_session.measure_amount[key.downcase]
+  end
+
   private
 
   def applicable_units
-    @applicable_units ||= ApplicableMeasureUnitMerger.new.call
+    @applicable_units ||= ApplicableMeasureUnitMerger.new(dedupe: false).call
```

Comment:
I think it's better to be more explicit:
`remove_duplication: true/false` , or `no_duplication`, or `without_duplicate`, ...
dedupe is not a word, isn't it?

Reply:
![Uploading Screenshot 2022-04-28 at 12.06.37.png…]()

Reply:
<img width="877" alt="Screenshot 2022-04-28 at 12 06 37" src="https://user-images.githubusercontent.com/58971/165739238-0f1c27d6-ce20-4c4d-8118-be8d675e7aa8.png">

Reply:
Spoke separately